library/core/src/fmt/mod.rs

```diff
@@ -316,7 +316,7 @@ impl<'a> Arguments<'a> {
     #[doc(hidden)]
     #[inline]
     #[unstable(feature = "fmt_internals", reason = "internal to format_args!", issue = "none")]
-    pub fn new_v1(pieces: &'a [&'static str], args: &'a [ArgumentV1<'a>]) -> Arguments<'a> {
+    pub const fn new_v1(pieces: &'a [&'static str], args: &'a [ArgumentV1<'a>]) -> Arguments<'a> {
```

Comment:
cc @rust-lang/const-eval or @RalfJung for new unstable const methods in this file.

Reply:
I cannot see a `const` user of this function, so why are you even changing this?

Reply:
Its uses are in macro-expanded code: 
https://github.com/rust-lang/rust/blob/00eefaccc5663107bc05a18c8c0df9799267ee95/src/test/pretty/issue-4264.pp#L34
and 
https://github.com/rust-lang/rust/blob/00eefaccc5663107bc05a18c8c0df9799267ee95/src/test/pretty/issue-4264.pp#L59

Reply:
Sure but where's the `const`? This macro cannot be used in a `const` AFAIK?

Reply:
Yeah, this is about const misconception we talk about.
Now it is only a hope that LLVM will see the const hint and optimize the code.